src/extension/src/scripts/generate-test.ts

```diff
@@ -0,0 +1,126 @@
+    setTimeout(resolve, time);
+  });
+};
+
+let attemptSync: any = (
+  instanceObj: CoreTemplateStudio,
+  syncAttemptNum: number
+) => {
+  if (syncAttemptNum < CONSTANTS.API.MAX_SYNC_REQUEST_ATTEMPTS) {
+    return instanceObj
+      .sync({
+        port: instance.getPort(),
+        payload: { path: CONSTANTS.API.DEVELOPMENT_PATH_TO_TEMPLATES },
+        liveMessageHandler: value => {
+          value;
+        }
+      })
+      .then(() => {
+        instance
+          .getFrameworks(projType)
+          .then(frameworks => {
+            frameworks.forEach(
+              (obj: { tags: { type: string }; name: string }) => {
+                if (obj.tags.type == "frontend") {
+                  frontends.push(obj.name);
+                } else if (obj.tags.type == "backend") {
+                  backends.push(obj.name);
+                }
+              }
+            );
+
+            let getPagesObj = (
+              instance: CoreTemplateStudio,
+              projType: string,
+              frontend: any,
+              backend: any
+            ) => {
+              return instance
+                .getPages(projType, frontend, backend)
+                .then(pages => {
+                  pagesObj = [];
+                  pages.forEach((page: { name: any; templateId: any }) => {
+                    pagesObj.push({
+                      name: page.name,
+                      identity: page.templateId
+                    });
+                  });
+                  return pagesObj;
+                })
+                .catch(err => {
+                  console.log(err);
+                });
+            };
+
+            let generateProj = (backend: string, frontend: string) => {
+              return getPagesObj(instance, projType, frontend, backend).then(
+                pagesObj => {
+                  return instance.generate({
+                    port: instance.getPort(),
+                    payload: {
+                      backendFramework: backend,
+                      frontendFramework: frontend,
+                      pages: pagesObj,
+                      path: "../../../../../src/extension/src/template_test",
+                      projectName: backend + "-" + frontend,
+                      projectType: projType,
+                      services: []
+                    },
+                    liveMessageHandler: value => {
+                      value;
+                    }
+                  });
+                }
+              );
+            };
+
+            let prevPromise: Promise<any> = Promise.resolve(null);
+            frontends.forEach(frontendFrameWork => {
+              backends.forEach(backendFramework => {
+                prevPromise = prevPromise.then(() =>
+                  generateProj(backendFramework, frontendFrameWork)
+                );
+              });
+            });
+
+            prevPromise.then(() => {
+              console.log("done");
```

Comment:
Instead of generating each permutation of framework one after the other, I think it could be faster and more readable to use a `Promise.all` instead. In that case, `prevPromise` would be an array of promises with different permutations of frontend and backend frameworks. Then we can use `Promise.all(prevPromise)` to resolve all the promises and say `done`.
Does that make sense or I am thinking something wrong?

Reply:
@njindal239 unfortunately, it is necessary that one generation of the framework happen after the other. This is because CoreTS gets corrupted if you try generating one project before another generation is done.

Reply:
@njindal239  Unfortunately, we must generate projects one after the other. This is because CoreTS gets corrupted if we try to generate a project before a previous generation is done. Thus, we must wait for one project to be generated before we continue to generate another. We achieve this by having a promise chain.

`Promise.all` would try to generate all projects at the same time (would send GENERATE commands one after the other before receiving replies from CoreTS through signalr)

Reply:
I agree. I did not know that we need to have one project complete its generation before we can start the second one. But that's a little weird to me why CoreTS behaves that way.

Reply:
The reason it works that way is because it was meant to be a 1 connection at a time thing, so it was developed limiting it to 1 connection. This can be changed fairly easily though if it proves to be a huge hinderance.